apis/v1alpha1/gateway_types.go

```diff
@@ -483,15 +476,17 @@ type GatewayStatus struct {
 	//
 	// These addresses should all be of type "IPAddress".
 	//
-	// +required
+	// +kubebuilder:validation:MaxItems=16
```

Comment:
Since `Addresses` is required, should an "Unknown" `AddressType` be supported for controllers that are unable to assign an address during reconciliation?

Reply:
// +kubebuilder:validation:MinItems=0 to match other optional lists.

Reply:
> Since Addresses is required, should an "Unknown" AddressType be supported for controllers that are unable to assign an address during reconciliation?

Shouldn't that be a Condition instead?

Reply:
I'm hesitant to add a minimum here since status will likely need to be populated at some point after a Gateway was created. As far as if a controller is unable to assign an address during reconciliation, I think that would be better communicated with some kind of Condition, maybe `Scheduled=False` with a reason of `NoResources`? Does that seem reasonable? If not, maybe we should continue this discussion in a follow up issue.